Client/Frontend/NotificationSurface/NotificationSurfaceManager.swift

```diff
@@ -71,7 +69,7 @@ class NotificationSurfaceManager: NotificationSurfaceDelegate {
         else { return }
 
         switch message.action {
-        case "OPEN_NEW_TAB":
+        case "://deep-link?url=homepanel/new-tab":
```

Comment:
I don't understand this.

The message action should map to either a deeplink or a web page. This seems to me that it only responds when `message.data.action == "OPEN_NEW_TAB"`, which is not right. Have I misunderstood this?

Reply:
So far we only have the mapping for `OPEN_NEW_TAB`. As soon as we have new notifications that should be scheduled we should be looking into a way of handling all the other deeplinks as well without adding to the `switch`. I think for now we could leave it like this. I can open a new task for this.

Reply:
The messaging system can be sending [any URL, including deeplinks] (https://experimenter.info/mobile-messaging/#list-of-actions) as actions, not just the URLs that are in the FML right now. Each surface should handle all actions.

`onMessagePressed` handles all the deeplinks. Please could you change this.